src/dishka/registry.py

```diff
@@ -320,12 +321,14 @@ def _process_decorator(
             )
         scope = self.dependency_scopes[provides]
         registry = self.registries[scope]
-        undecorated_type = NewType(
+        undecorated_type = NewType(  # type: ignore[misc]
             f"{provides.type_hint.__name__}@{self.decorator_depth[provides]}",
-            decorator.provides.type_hint,
+            decorator.provides.type_hint,  # type: ignore[name-defined]
         )
         self.decorator_depth[provides] += 1
         old_factory = registry.get_factory(provides)
+        if old_factory is None:
+            raise NoFactoryError(provides)
```

Comment:
Let's have `InvalidGraphError` here with more descpriptive text. Something like: `cannot apply decorator because there is no factory for {provides}`